:broom: (#1906)
enkidevs/curriculum: javascript/core/objects/object-part-1.md

There are a couple of ways to create objects in JavaScript.

Probably the most common is with the object initializer:

```js

> let per = {

Author: ```suggestion
let person = {
```

These properties are generally accessed via dot notation.

> We can access our person objects name with the dot syntax:

Author: ```suggestion
We can access our `person` object's name with the dot syntax:
```

```js

> console.log(per.firstName); // "john"

Author: ```suggestion
console.log(person.firstName); // "john"
```

```

We can also access properties using the array syntax.

> This can be useful when you do not know the name of the property you want to change it at runtime dynamically .

Reviewer: ```suggestion
This can be very useful, especially when you do not know the name of the property you want to dynamically change at runtime .
```

```js
let propName = "firstName";

> console.log(per[propName]);

Author: ```suggestion
console.log(person[propName]);
```

```

We can even add properties anytime we want to our objects:

```js
// using dot notation
per.age = "40";

// using bracket notation

> per["homeTown"] = "London";

Author: ```suggestion
person.age = "40";

// using bracket notation
person["homeTown"] = "London";
```

## Revision

Which line adds a new property to an object that looks like this:

```js

> var per = {

Author: ```suggestion
let person = {
```

  firstName = "John",
  lastName = "Smith"
};

???
```

* `per.age = 40;`
* `var per.age = 40;`

> * `per[age] = 40;`

Author: ```suggestion
* `person.age = 40;`
* `let per.age = 40;`
* `person[age] = 40;`
```

> * `per.addNewProperty(age, 40);`

Author: ```suggestion
* `person.addNewProperty(age, 40);`
```
